pageserver/src/deletion_queue.rs

```diff
@@ -715,6 +713,10 @@ impl DeletionQueue {
                 tracing::warn!("Timed out flushing deletion queue on shutdown")
             }
         }
+
+        // We only cancel _after_ flushing: otherwise we would be shutting down the
+        // components that do the flush.
+        self.cancel.cancel();
```

Comment:
Interesting that deletion queue has a sort of similar issue as with the remote timeline client that it needs to be detached from the whole tree because we use it during shutdown. Noted re: #6134.